solr/contrib/ltr/src/java/org/apache/solr/ltr/feature/OriginalScoreFeature.java

```diff
@@ -72,14 +72,16 @@ public String toString() {
     public FeatureScorer scorer(LeafReaderContext context) throws IOException {
 
       final Scorer originalScorer = w.scorer(context);
-      return new OriginalScoreScorer(this, originalScorer);
+      return new OriginalScoreScorer(this, originalScorer,
+          DocIdSetIterator.all(DocIdSetIterator.NO_MORE_DOCS));
     }
 
     public class OriginalScoreScorer extends FeatureScorer {
       Scorer originalScorer;
 
-      public OriginalScoreScorer(FeatureWeight weight, Scorer originalScorer) {
-        super(weight);
+      public OriginalScoreScorer(FeatureWeight weight, Scorer originalScorer,
+          DocIdSetIterator itr) {
```

Comment:
In this case I wouldn't pass in the iter in the constructor since we use the scorers iterator anyways.  You can call super(weight, originalScorer.iterator())  and possibly delete the local iterator() function?